From 7a1310612851ad27fc7ca07238f683737f388344 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?I=C3=B1igo=20Mart=C3=ADnez?= <inigomartinez@gmail.com>
Date: Thu, 28 Sep 2017 16:27:17 +0200
Subject: [PATCH] project: Fix documentation generation

Documentation generation with GtkDoc is failing due to missing
directories.

This is happening because GtkDoc tries to search for directories
that are present in the source code tree, but are not in the build
tree: 'src/engine', 'src/interfaces' and 'src/notification'.

This patch modifies the directories that GtkDoc scans in search for
source code files.

https://bugzilla.gnome.org/show_bug.cgi?id=788295
---
 doc/reference/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/reference/meson.build b/doc/reference/meson.build
index 46b1f76..6d25d0b 100644
--- a/doc/reference/meson.build
+++ b/doc/reference/meson.build
@@ -53,7 +53,7 @@ doc_path = join_paths(gnome_todo_datadir, 'gtk-doc', 'html', meson.project_name(
 gnome.gtkdoc(
   meson.project_name(),
           main_sgml: meson.project_name() + '-docs.sgml',
-            src_dir: src_inc,
+            src_dir: join_paths(meson.source_root(), 'src'),
        dependencies: libgtd_dep,
           mkdb_args: [ '--sgml-mode', '--output-format=xml' ],
         html_assets: html_images,
